[asio] Prevent `add_library(asio::asio)` if target already exists.

****

**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]
Following CMakeLists.txt fails:
```
# Find twice
find_package(asio CONFIG REQUIRED)
find_package(asio CONFIG REQUIRED)
```
with error:
```
[cmake] CMake Error at /opt/vcpkg/scripts/buildsystems/vcpkg.cmake:618 (_add_library):
[cmake]   _add_library cannot create imported target "asio::asio" because another
[cmake]   target with the same name already exists.
[cmake] Call Stack (most recent call first):
[cmake]   /opt/vcpkg/installed/x64-linux/share/asio/asio-config.cmake:3 (add_library)
[cmake]   /opt/vcpkg/scripts/buildsystems/vcpkg.cmake:829 (_find_package)
[cmake]   CMakeLists.txt:25 (find_package)
```

**Proposed solution**
Do `add_library` only if target `asio::asio` is not already exists.

--- a/ports/asio/asio-config.cmake
+++ b/ports/asio/asio-config.cmake
@@ -1,8 +1,8 @@
 include ("${CMAKE_CURRENT_LIST_DIR}/asio-targets.cmake")
 
 if(NOT TARGET asio::asio)
-	add_library(asio::asio INTERFACE IMPORTED)
-	target_link_libraries(asio::asio INTERFACE asio)
+    add_library(asio::asio INTERFACE IMPORTED)
+    target_link_libraries(asio::asio INTERFACE asio)
 endif()
 
 get_target_property(_ASIO_INCLUDE_DIR asio INTERFACE_INCLUDE_DIRECTORIES)

--- a/ports/asio/asio-config.cmake
+++ b/ports/asio/asio-config.cmake
@@ -1,6 +1,9 @@
 include ("${CMAKE_CURRENT_LIST_DIR}/asio-targets.cmake")
-add_library(asio::asio INTERFACE IMPORTED)
-target_link_libraries(asio::asio INTERFACE asio)
+
+if(NOT TARGET asio::asio)
+	add_library(asio::asio INTERFACE IMPORTED)
+	target_link_libraries(asio::asio INTERFACE asio)
+endif()
 
 get_target_property(_ASIO_INCLUDE_DIR asio INTERFACE_INCLUDE_DIRECTORIES)
 set(ASIO_INCLUDE_DIR "${_ASIO_INCLUDE_DIR}")

--- a/ports/asio/vcpkg.json
+++ b/ports/asio/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "asio",
   "version": "1.24.0",
+  "port-version": 1,
   "description": "Asio is a cross-platform C++ library for network and low-level I/O programming that provides developers with a consistent asynchronous model using a modern C++ approach.",
   "homepage": "https://github.com/chriskohlhoff/asio",
   "documentation": "https://think-async.com/Asio/asio-1.24.0/doc/",

--- a/versions/a-/asio.json
+++ b/versions/a-/asio.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "b134a3e21a2ef661aa5e3802cefc22386c095aaa",
+      "version": "1.24.0",
+      "port-version": 1
+    },
+    {
       "git-tree": "bba8740d419878e427c71f076d569f8a26833c6b",
       "version": "1.24.0",
       "port-version": 0

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -198,7 +198,7 @@
     },
     "asio": {
       "baseline": "1.24.0",
-      "port-version": 0
+      "port-version": 1
     },
     "asio-grpc": {
       "baseline": "2.3.0",

